[CI] MixedClusterEsqlSpecIT test {inlinestats.BeforeKeep ASYNC} failing

**Build Scans:**
- [elasticsearch-intake #8489 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/exqvp3tblw7g4)
- [elasticsearch-pull-request #26727 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/kivh3ltxtq6v6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.16.0#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT" -Dtests.method="test {inlinestats.BeforeKeep ASYNC}" -Dtests.seed=2D13F745760D338B -Dtests.bwc=true -Dtests.locale=pt -Dtests.timezone=America/Mazatlan -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Binlinestats.BeforeKeep%20ASYNC%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Data mismatch:
row 0 column 0:a list containing
row 0 column 0:0: expected "10001" but was "10066"
row 0 column 1:a list containing
row 0 column 1:0: expected "2" but was "5"
row 0 column 2:a list containing
row 0 column 2:0: expected "73578" but was "66817"
row 1 column 0:a list containing
row 1 column 0:0: expected "10002" but was "10067"
row 1 column 1:a list containing
row 1 column 1:0: expected "5" but was "2"
row 1 column 2:a list containing
row 1 column 2:0: expected "66817" but was "73578"
row 2 column 0:a list containing
row 2 column 0:0: expected "10003" but was "10068"
row 2 column 1:a list containing
row 2 column 1:0: expected "4" but was "3"
row 2 column 2:a list containing
row 2 column 2:0: expected "74572" but was "74970"
```

**Issue Reasons:**
- [main] 2 failures in test test {inlinestats.BeforeKeep ASYNC} (28.6% fail rate in 7 executions)
- [main] 2 failures in step 8.16.0_bwc-snapshots (28.6% fail rate in 7 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -108,18 +108,6 @@ tests:
 - class: org.elasticsearch.repositories.azure.AzureBlobContainerRetriesTests
   method: testReadNonexistentBlobThrowsNoSuchFileException
   issue: https://github.com/elastic/elasticsearch/issues/111233
-- class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-  method: test {inlinestats.BeforeKeep ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/111257
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {inlinestats.BeforeKeep ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/111259
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {inlinestats.BeforeKeep SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/111260
-- class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-  method: test {inlinestats.BeforeKeep SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/111262
 - class: org.elasticsearch.xpack.esql.qa.single_node.RestEsqlIT
   method: testInlineStatsProfile {SYNC}
   issue: https://github.com/elastic/elasticsearch/issues/111263

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/inlinestats.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/inlinestats.csv-spec
@@ -336,18 +336,16 @@ emp_no:integer | languages:integer | max_salary:integer
          10003 |                 4 | 74572
 ;
 
-beforeKeep
+beforeKeepWhere
 required_capability: inlinestats
 
 FROM employees
 | INLINESTATS max_salary = MAX(salary) by languages
 | KEEP emp_no, languages, max_salary
-| LIMIT 3;
+| WHERE emp_no == 10003;
 
 ignoreOrder:true
 emp_no:integer | languages:integer | max_salary:integer
-         10001 |                 2 | 73578
-         10002 |                 5 | 66817
          10003 |                 4 | 74572
 ;
 

